src/main/java/com/nicoislost/ZoomO.java

```diff
@@ -28,197 +45,154 @@ public void onInitializeClient() {
 	}
 
 	/**
-	 * Tells us how many keybindings are pressed to use in
-	 * {@link #zoomModifierNum()}
-	 * 
-	 * @return 0-3 depending on how many keys are pressed.
+	 * Returns the configured zoom modifier for the currently active key binding
+	 * @return The configure zoom modifier value for the active key binding or 0 if no key binding is active
 	 */
-	public static int keysPressed() {
-		//This converts our booleans into ints and sums them up.
-		return (key1() ? 1 : 0) + (key2() ? 1 : 0) + (key3() ? 1 : 0);
+	private static double getCurrentZoomModifier() {
+		KeyBind keyBind = getCurrentKeyBind();
+		if (keyBind == null) {
+			return 0;
+		}
+		LOGGER.info("Current key = " + keyBind);
+		return keyBind.getConfigValue();
 	}
 
 	/**
-	 * @return 1 if 0 or 1 keys are pressed, 2 if 2 keys are pressed, 6 if 3 keys
-	 *         are pressed.
+	 * Delegates the functionality of the MixIn callback
+	 * @param callbackInfo
 	 */
-	public static int keyNum() {
-		int num = 1;
-		// We start with one to avoid multiplying by zero, and it is only activates if
-		// at least 1 key is pressed.
+	// One thing to remember is, this is going to be called ALOT while one of the key bindings is being pressed,
+	// so we should be conscious of doing as little work as fast as possible - so, caching values if possible and/or
+	// reducing the amount of computational work we're doing.
+	// One thing we could do is check the previous state with the current state and if nothing's changed, returned
+	// the previously calculated value, but, arguably, comparing two `List` isn't the fastest thing in the world either,
+	// so maybe a `BitSet` instead?
+	public static void getZoomLevel(CallbackInfoReturnable<Double> callbackInfo) {
+		syncKeys();
+		if (isZooming()) {
+			double fov = callbackInfo.getReturnValue();
+			callbackInfo.setReturnValue(fov * ZoomO.getZoomFactor());
+		}
+		ZoomO.smooothDuuude();
+	}
 
-		if (key2()) {
-			num = num * 2;
+	/**
+	 * Calculates the zoom factor to be applied based on the current active key binding and it's configured zoom
+	 * state
+	 * @return The zoom factor for the key binding or `1` if no zooming is occurring
+	 */
+	public static double getZoomFactor() {
+		// Insert some paranoia
+		if (!isZooming()) {
+			LOGGER.info("You're not zooming, what are you doing here?!");
+			return 1;
 		}
-		if (key3()) {
-			num = num * 3;
+		double modifier = getCurrentZoomModifier();
+		LOGGER.info("modifier = " + modifier);
+		return 1 - (modifier / 100);
+	}
+
+	private static boolean isSmoothCameraForCurrentKeyEnabled() {
+		KeyBind keyBind = getCurrentKeyBind();
+		if (keyBind == null) {
+			return false;
 		}
-		return num;
+		return keyBind.isSmoothCameraEnabled();
+//		boolean enabled = false;
+//		switch (keyBind) {
+//			case LEVEL_1 -> enabled = CONFIG.zoom1SmoothCamera();
+//			case LEVEL_2 -> enabled = CONFIG.zoom2SmoothCamera();
+//			case LEVEL_3 -> enabled = CONFIG.zoom3SmoothCamera();
+//		}
+//		return enabled;
```

Comment:
None of the commented code is actually needed any more, but demonstrates the neat idea of binding the config and key bindings together